src/hotspot/share/oops/flatArrayOop.hpp

```diff
@@ -29,7 +29,8 @@
 #include "oops/klass.hpp"
 #include "runtime/handles.hpp"
 
-// A flatArrayOop is an array containing inline types (may include flatten embedded oop elements).
+// A flatArrayOop is an array containing flattened inline types (no indirection).
```

Comment:
This is one aspect of the renaming that requires some bike-shading.
So far, I've applied the "flat" qualifier to the container, the field or the array, because there's a direct impact on the shape of the container and the way it is accessed.
But should we apply the "flat" qualifier to the value itself? The value is the value, a set of constants with a shape that doesn't change if it is stored in a flat field, a flat array or an standalone instance.
Both "flat inline type" and "flattened inline type" sound weird to me, so I don't really have a preference. Does one make more sense or is more informative than the other?

Reply:
I don't have a strong opinion but maybe we should simply rephrase this to:
`A flatArrayOop is a flat array containing inline types (no indirection).`

Or

`A flatArrayOop points to a flat array containing inline types (no indirection).`